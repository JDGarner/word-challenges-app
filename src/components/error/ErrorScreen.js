import React from "react";
import { View } from "react-native";
import styled from "styled-components";
import { MediumText, LargeText } from "../text/Text";
import { PaddedButton } from "../button/Button";
import { ScreenContainerPadded } from "../containers/Containers";
import TopBar from "../top-bar/TopBar";

export const ERROR_CODES = {
  CONNECTION: "CONNECTION",
  TIMEOUT: "TIMEOUT",
  API: "API",
  GENERIC: "GENERIC",
};

const ERROR_CODES_TO_TEXT = {
  CONNECTION: "Network error, no internet connection",
  TIMEOUT: "Network request timed out",
  API: "Oops, something went wrong, our servers may be down at the moment",
  GENERIC: "Oops, something went wrong, our servers may be down at the moment",
};

const ErrorText = styled(MediumText)`
  text-align: center;
  margin-bottom: 30px;
`;

const ErrorButton = styled(PaddedButton)`
  width: 160;
  justify-content: center;
  align-items: center;
`;

const ErrorScreen = ({ onButtonPress, errorCode, onPressBack }) => {
  return (
    <ScreenContainerPadded>
      {/* <TopBar onPressLeftButton={onPressBack} /> */}
      <View
        style={{
          flex: 1,
          marginBottom: 120,
          justifyContent: "center",
          alignItems: "center",
          width: "100%",
        }}>
        <ErrorText>{ERROR_CODES_TO_TEXT[errorCode]}</ErrorText>
        {onButtonPress && (
          <ErrorButton onPress={onButtonPress}>
            <LargeText>Retry</LargeText>
          </ErrorButton>
        )}
      </View>
    </ScreenContainerPadded>
  );
};

ErrorScreen.defaultProps = {
  errorCode: ERROR_CODES.GENERIC,
};

export default ErrorScreen;
