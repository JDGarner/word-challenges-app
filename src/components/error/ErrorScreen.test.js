import "react-native";
import React from "react";
import { fireEvent } from "@testing-library/react-native";
import ErrorScreen from "./ErrorScreen";
import { renderWithTheme } from "../../testing";

describe("Error Screen", () => {
  it("should match snapshot", () => {
    const mockFn = jest.fn();

    const rendered = renderWithTheme(
      <ErrorScreen errorCode="GENERIC" onButtonPress={mockFn} />,
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it("fires the error screen button press", () => {
    const mockFn = jest.fn();

    const { getByText } = renderWithTheme(
      <ErrorScreen errorCode="GENERIC" onButtonPress={mockFn} />,
    );

    fireEvent.press(getByText("Retry"));

    expect(mockFn).toBeCalled();
  });
});
