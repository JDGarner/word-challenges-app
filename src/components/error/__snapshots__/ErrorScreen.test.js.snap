// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Screen should match snapshot 1`] = `
<View
  className="sc-gzVnrw sc-jTzLTM gkVpDJ"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 120,
        "width": "100%",
      }
    }
  >
    <Text
      className="sc-ifAKCX sc-kgoBCf eoSCgB"
    >
      Oops, something went wrong, our servers may be down at the moment
    </Text>
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        className="sc-gqjmRU bsUHzu"
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={Array []}
      >
        <Text
          className="sc-bwzfXH etwhaW"
        >
          Retry
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ErrorScreen should match snapshot 1`] = `
<View
  className="sc-gzVnrw sc-jTzLTM gkVpDJ"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 120,
        "width": "100%",
      }
    }
  >
    <Text
      className="sc-ifAKCX sc-kgoBCf eoSCgB"
    >
      Oops, something went wrong, our servers may be down at the moment
    </Text>
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        className="sc-gqjmRU bsUHzu"
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={Array []}
      >
        <Text
          className="sc-bwzfXH etwhaW"
        >
          Retry
        </Text>
      </View>
    </View>
  </View>
</View>
`;
